libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -474,16 +475,30 @@ class MapboxNavigation(
      *
      * If the list is empty, the SDK will exit the `Active Guidance` state.
      *
+     * Note: use [cleanRoutes] to exit `Active Guidance` state.
+     *
      * Use [RoutesObserver] and [MapboxNavigation.registerRoutesObserver] to observe whenever the routes list reference managed by the SDK changes, regardless of a source.
      *
      * @param routes a list of [DirectionsRoute]s
      * @see [requestRoutes]
      */
-    fun setRoutes(routes: List<DirectionsRoute>) {
+    fun setRoutes(routes: List<DirectionsRoute>, origin: RouterOrigin) {
```

Comment:
Why is the `origin` needed to `setRoutes`? Also, couldn't the `routes` be a mix from different `Router`s? 🤔

Reply:
> Why is the origin needed to setRoutes?

to spread origin via `RoutesObserver`. 
for instance it useful for reroute: reroute happens under the hood of the SDK and the only way to know where the route has fetched (offboard/onboard) it's listening for `RoutesObserver`

> Also, couldn't the routes be a mix from different Routers? 

from Nav SDK it's `offboard` or `onboard` only, but if a customer wants to use a different router, for that they must use `RoutersOrigin.Custom`

Reply:
It does not look like the origin matters here though - it doesn't do anything. 

Or do we want to add a `RouterOrigin` type, which is for the "bring your own route"?

Reply:
> It does not look like the origin matters here though - it doesn't do anything.

No, it does not. Just provide to listeners where the route was fetched from 

> do we want to add a RouterOrigin type, which is for the "bring your own route"?

I think it would be useful, for that we have `Custom` `RouterOrigin`.

Reply:
> > Why is the origin needed to setRoutes?

>to spread origin via RoutesObserver.
for instance it useful for reroute: reroute happens under the hood of the SDK and the only way to know where the route has fetched (offboard/onboard) it's listening for RoutesObserver

As far as I can tell, this is the only practical use case for also requiring the origin with `MapboxNavigation#setRoutes` and it feels like a workaround. Is there a way for us to resurface the route origin information via `RerouteController`  better instead of adding it to `MapboxNavigation#setRoutes` and the `RoutesObserver`? Interested developers could override some of the `RerouteControllers` methods to get access to this information.

Reply:
make sense, need a new round to fix that

Reply:
Yeah, I'd love to implement something around these lines - especially because with the current approach we'd be forcing developers to provide an `origin` (that should be transparent IMO) + adding a new API `cleanRoutes` (which might add confusion and complicates `MapboxNavigation`'s lifecycle).

Reply:
fixed